pywemo/subscribe.py

```diff
@@ -34,33 +72,138 @@ def get_ip_address(host='1.2.3.4'):
+      from the pywemo SubscriptionRegistry. The event type for a long press
+      action is EVENT_TYPE_LONG_PRESS.
+    """
+
+    def do_NOTIFY(self):  # pylint: disable=invalid-name
         """Handle subscription responses received from devices."""
         sender_ip, _ = self.client_address
         outer = self.server.outer
         device = outer.devices.get(sender_ip)
-        content_len = int(self.headers.get('content-length', 0))
-        data = self.rfile.read(content_len)
         if device is None:
             LOG.warning('Received event for unregistered device %s', sender_ip)
         else:
-            # trim garbage from end, if any
-            data = data.decode("UTF-8").split("\n\n")[0]
-            doc = cElementTree.fromstring(data)
+            doc = self._get_xml_from_http_body()
             for propnode in doc.findall('./{0}property'.format(NS)):
                 for property_ in list(propnode):
                     text = property_.text
                     outer.event(device, property_.tag, text)
 
-        self.send_response(200)
-        self.send_header('Content-Type', 'text/html')
-        self.send_header('Content-Length', len(SUCCESS))
+        self._send_response(200, RESPONSE_SUCCESS)
+
+    def do_GET(self):  # pylint: disable=invalid-name
+        """Handle GET requests for a Virtual WeMo device."""
+        if self.path.endswith("/setup.xml"):
+            self._send_response(
+                200, VIRTUAL_SETUP_XML, content_type="text/xml"
+            )
+        else:
+            self._send_response(404, RESPONSE_NOT_FOUND)
+
+    def do_POST(self):  # pylint: disable=invalid-name
+        """Handle POST requests for a Virtual WeMo device."""
+        if self.path.endswith("/upnp/control/basicevent1"):
+            sender_ip, _ = self.client_address
+            outer = self.server.outer
+            device = outer.devices.get(sender_ip)
+            if device is None:
+                LOG.warning(
+                    'Received event for unregistered device %s', sender_ip
+                )
+            else:
+                doc = self._get_xml_from_http_body()
+                binary_state = doc.find('.//BinaryState')
+                if binary_state is not None:
+                    text = binary_state.text
+                    outer.event(device, EVENT_TYPE_LONG_PRESS, text)
+            self._send_response(200, RESPONSE_SUCCESS)
+        else:
+            self._send_response(404, RESPONSE_NOT_FOUND)
+
+    def do_SUBSCRIBE(self):  # pylint: disable=invalid-name
+        """Handle SUBSCRIBE requests for a Virtual WeMo device."""
+        if self.path.endswith("/upnp/event/basicevent1"):
+            self.send_response(200)
+            self.send_header("CONTENT-LENGTH", "0")
+            self.send_header("TIMEOUT", "Second-1801")
+            self.send_header(
+                "SID", "uuid:a74b23d5-34b9-4f71-9f87-bed24353f304"
```

Comment:
Maybe add a comment here or at the top of this method about a random valid UUID being selected just to avoid any potential confusion here in the future

Reply:
Added.

By the way, I added your `build.sh` script as a pre-commit git hook. That's been super helpful in catching mistakes before they are committed.

Reply:
Ya, I've been thinking about having the build.sh file install a pre-commit git hook.  Or maybe separate the file into 2 parts, one that installs the libraries, poetry, and the pre-commit hook and a second that simply runs the various checks (which is the one that would be run for the pre-commit hook).

Reply:
added a new issue to act as a reminder